pkg/engine/virtualmachinescalesets.go

```diff
@@ -575,16 +579,19 @@ func CreateAgentVMSS(cs *api.ContainerService, profile *api.AgentPoolProfile) Vi
 
 	outBoundCmd := ""
 	registry := ""
-
+	ncBinary := "nc"
+	if cs.Properties.MasterProfile != nil && cs.Properties.MasterProfile.IsCoreOS() {
```

Comment:
So we actually want a agentPool-specific method here, as this vmss object constructor is per-agentPool. So we'd replace `cs.Properties.MasterProfile.IsCoreOS()` with `profile.IsCoreOS()`.